src/rpg_database/character_sheet/FocusesSet.java

```diff
@@ -0,0 +1,136 @@
+package rpg_database.character_sheet;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import rpg_database.character_sheet.interfaces.CustomSetter;
+
+public class FocusesSet implements Set<FocusesLogic>, CustomSetter<FocusesSet> {
+
+	private final HashSet<FocusesLogic> focusesSet;
+
+	public FocusesSet(FocusesLogic... focuses) {
+		focusesSet = new HashSet<FocusesLogic>();
+		ArrayList<FocusesLogic> focusList = findMultipleTimesAddedFocusesInTheParameterList(focuses);
+		focusesSet.addAll(focusList);
+	}
+
+	@Override
+	public Class<FocusesSet> getImplementingClass() {
+		return FocusesSet.class;
+	}
+
+	@Override
+	public void setSelfInSheet(CharacterSheet characterSheet) {
+		characterSheet.characterData.put(Fields.FOCUSES, this);
+	}
+
+	@Override
+	public int size() {
+		return focusesSet.size();
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return focusesSet.isEmpty();
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		return focusesSet.contains(o);
+	}
+
+	@Override
+	public Iterator<FocusesLogic> iterator() {
+		return focusesSet.iterator();
+	}
+
+	@Override
+	public Object[] toArray() {
+		return focusesSet.toArray();
+	}
+
+	@Override
+	public <T> T[] toArray(T[] a) {
+		return focusesSet.toArray(a);
+	}
+
+	@Override
+	public boolean add(FocusesLogic e) {
+		return focusesSet.add(e);

```

Comment:
itt nem kellene csekkolni előbb hogy a FocusesLogic objektumoknak ugyanaz -e a focus adattagja? Mi van ha csinálok két FocusesLogic objectet:
1. focus == MIGHT, focusImprovement == 0,
2. focus == MIGHT, focusImprovement == 1.
Az elsőt a FocusesSet inicializálásánál átadtam, a másodikat pedig addolom hozzá később.
Ekkor ugye mivel nem lesz egyenlő a két objektum (ugye a Set hívja az objektumok equals metódusát és az alapján szűri ki a duplikátumokat) ezért két MIGHT-os objektum is be fog kerülni, különböző focusImprovement értékekkel a listába. és sosem fogjuk tudni megfejteni belőle hogy a user (vagy az ezt hívó kódrészlet) eredetileg mi a szent szart akarhatott vele.